src/app/ad-authoring/ad-authoring.component.ts

```diff
@@ -25,6 +26,19 @@ export class AdAuthoringComponent {
   LandingTypeMapping = LANDING_TYPE_DISPLAY_VALUES;
   landingTypeValues = sortedLandingType;
 
+  validUrlRegex = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)?/gi;
+  url = new FormControl('', [
+    Validators.required,
+    Validators.pattern(this.validUrlRegex),
+  ]);
+
+  getErrorMessage() {
+    if (this.url.hasError('required')) {
+      return 'You must enter a value';
+    }
+    return 'Invalid URL';
```

Comment:
We can use a ternary here and drop the if.

```suggestion
    return this.url.hasError('required') ?  'You must enter a value' :  'Invalid URL';
```